### DISCUSSION TITLE
How to import traces from a .txt or .csv file ?

### DISCUSSION THREAD

**Discussion by sourav003** [omnetpp/omnetpp#1196] [Q&A]
Hi
This question may be trivial to many of you but I am asking this being a newbee to OMNeT++. Apologies in advance.
I have file (.txt/.csv) containing a column of floating point numbers as follows:
`0.0034158
0.0016494
0.034401
0.0015104
0.0076398
0.038799
0.00059583
0.0308
0.00049757
0.00072977
0.01205`
These can be considered as the time differences between consecutive packet transmissions. So, I need to pass the path of the file as an input parameterin the NED file to some simple module and then store it in a vector and access the consecutive elements in the runtime in the .cc file.
So, it will be extremely helpful if anyone can help me with this basic code.

Thanks in advance.

**Comment by aarizaq** [comment]
I think that you will need to program a parser in the c++ module that process the file. 

If the file is organized in lines as
0.0034158 0.0016494 0.034401 0.0015104 0.0076398 0.038799 0.00059583 0.0308 0.00049757 0.00072977 0.01205
0.0034158 0.0016494 0.034401 0.0015104 0.0076398 0.038799 0.00059583 0.0308 0.00049757 0.00072977 0.01205
0.0034158 0.0016494 0.034401 0.0015104 0.0076398 0.038799 0.00059583 0.0308 0.00049757 0.00072977 0.01205


    std::string fi(fileName); // only the file name
    fi = "./"+fi; // this include the actual directory
    std::ifstream inFile(fi.c_str());

    if (!inFile)
    {
        error("Error while opening input file (File not found or incorrect type)\n");
    }
   std::vector<double> values;
    while (!inFile.eof())
    {
        std::string line;
        std::getline (inFile,line); // read a line
        std::vector<std::string> v =cStringTokenizer.html)(line).asVector(); // parse the line
       for (const auto &e: v) { // convert the strings to double and store it in the vector
         try {
             double val = std::stod(e);
             values.push_back(val)
         }
         catch (const std::invalid_argument& ia) {
	        std::cerr << "Invalid argument: " << ia.what() << '\n';
          }
         catch (const std::out_of_range& oor) {
            std::cerr << "Out of Range error: " << oor.what() << '\n';
         }
       }
    }
    // if you want to ordered the vector.
    std::sort(values.begin(), values.end());
